bin/common/write_k8s_ip_to_environment.sh

```diff
@@ -0,0 +1,8 @@
+#!/usr/bin/env bash
+
+set -o errexit
+while ! ifconfig eth1 | grep -q addr; do
+  sleep 3
+done
+K8S_VM_IP=$(ifconfig eth1 | grep -oE 'addr:[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}' | sed 's/addr://g')
+echo K8S_VM_IP=$K8S_VM_IP >> /etc/environment
```

Comment:
If we run the cert generation in ExecStartPre, I'm pretty sure that will break already-existing deployments between restarts. Do you know if we actually *need* the IP to go into the certificate? At least right now, since apiserver binds to 127.0.0.1 insecurely (but surprisingly actually binds to 0.0.0.0), I'm not sure if we need the cert to contain the 'public' IP. I think the internal communication happens over one of the k8s internal domains.